wfe2/wfe.go

```diff
@@ -1250,3 +1253,40 @@ func (wfe *WebFrontEndImpl) NewOrder(ctx context.Context, logEvent *requestEvent
 		return
 	}
 }
+
+// Order is used to retrieve a existing order object
+func (wfe *WebFrontEndImpl) Order(ctx context.Context, logEvent *requestEvent, response http.ResponseWriter, request *http.Request) {
+	id, err := strconv.ParseInt(request.URL.Path, 10, 64)
+	if err != nil {
+		wfe.sendError(response, logEvent, probs.Malformed("Invalid ID"), err)
+		return
+	}
+
+	order, err := wfe.SA.GetOrder(ctx, &sapb.OrderRequest{Id: &id})
+	if err != nil {
+		if berrors.Is(err, berrors.NotFound) {
+			wfe.sendError(response, logEvent, probs.NotFound("No order for ID"), err)
```

Comment:
Nit: Can you include `id` in in the detail message so its easier to see which order ID wasn't found?